zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/KerberosInterpreter.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.interpreter;
+
+import java.util.Properties;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.zeppelin.annotation.ZeppelinApi;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Interpreter wrapper for Kerberos initialization
+ *
+ * TODO: coming soon
+ * Please see,
+ * https://zeppelin.apache.org/docs/latest/development/writingzeppelininterpreter.html
+ *
+ * runKerberosLogin() method you need to implement that determine Zeppelin's behavior.
+ * startKerberosLoginThread() needs to be called inside the open() and
+ * shutdownExecutorService() inside close().
+ */
+public abstract class KerberosInterpreter extends Interpreter {
+
+  Integer kinitFailCount = 0;
```

Comment:
perhaps a child class wants to check this? we should report this in the error if an operation fails because the token can't be refreshed in the background?